src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -599,6 +599,21 @@ export const GlobalProvider = memo(
             [changeNodes, changeEdges]
         );
 
+        // TODO: This shouldn't be necessary, but it at least fixes the issue
```

Comment:
So why is it necessary?

Reply:
Deleting an iterator doesn't delete an edge if the edge is connected to an outer node. I couldn't think of a way we could do this in removeNodeById, so I just did it here

Reply:
Ah, I forget to account for iterator helper nodes in the code snippet, but the general idea should be clear.